chainercv/extensions/evaluator/semantic_segmentation_evaluator.py

```diff
@@ -0,0 +1,109 @@
+import copy
+import numpy as np
+
+from chainer import reporter
+import chainer.training.extensions
+
+from chainercv.evaluations import eval_semantic_segmentation
+from chainercv.utils import apply_prediction_to_iterator
+
+
+class SemanticSegmentationEvaluator(chainer.training.extensions.Evaluator):
+
+    """An extension that evaluates a semantic segmentation model.
+
+    This extension iterates over an iterator and evaluates the prediction
+    results of the model by common evaluation metrics for semantic
+    segmentation.
+    This extension reports values with keys below.
+    Please note that :obj:`'iou/<label_names[l]>'` and
+    :obj:`'class_accuracy/<label_names[l]>'` are reported only if
+    :obj:`label_names` is specified.
+
+    * :obj:`'miou'`: Mean of IoUs (mIoU).
+    * :obj:`'iou/<label_names[l]>'`: IoU for class \
+        :obj:`label_names[l]`, where :math:`l` is the index of the class. \
+        For example, if :obj:`label_names` is \
+        :obj:`~chainercv.datasets.camvid_label_names`, \
+        this evaluator reports :obj:`'iou/Sky'`, \
+        :obj:`'ap/Building'`, etc.
+    * :obj:`'mean_class_accuracy'`: Mean of class accuracies.
+    * :obj:`'class_accuracy/<label_names[l]>'`: Class accuracy for class \
+        :obj:`label_names[l]`, where :math:`l` is the index of the class.
+    * :obj:`'pixel_accuracy'`: Pixel accuracy.
+
+    If there is no label assigned to class :obj:`label_names[l]`
+    in the ground truth, values corresponding to keys
+    :obj:`'iou/<label_names[l]>'` and :obj:`'class_accuracy/<label_names[l]>'`
+    are :obj:`numpy.nan`.
+    In that case, the means of them are calculated by excluding them from
+    calculation.
+
+    For details on the evaluation metrics, please see the documentation
+    for :func:`chainercv.evaluations.eval_semantic_segmentation`.
+
+    .. seealso::
+        :func:`chainercv.evaluations.eval_semantic_segmentation`.
+
+    Args:
+        iterator (chainer.Iterator): An iterator. Each sample should be
+            following tuple :obj:`img, label`.
+            :obj:`img` is an image, :obj:`label` is pixel-wise label.
+        target (chainer.Link): A semantic segmentation link. This link should
+            have :meth:`predict` method which takes a list of images and
```

Comment:
`which` -> `that`? https://github.com/chainer/chainercv/pull/229#discussion_r119510618